vue/src/modules/Workspace/article/ArticleForm.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <ValidationObserver ref="form" v-slot="{ handleSubmit, invalid ,reset}">
+    <b-modal
+      :visible="showModal" id="article-form" ref="modal" size="lg" :title="modalTitle"
+      @hidden="hideModal" @ok.prevent="handleSubmit(onSubmit)" :ok-title="$t('Submit')" scrollable>
+      <b-form @submit.prevent="handleSubmit(onSubmit)" novalidate>
+        <input-widget
+          :model="model" attribute="image" type="file" :placeholder="$t('Choose a image or drop it here...')">
+        </input-widget>
+        <input-widget :model="model" attribute="title"></input-widget>
+        <input-widget :model="model" attribute="body" type="richtext"></input-widget>
+      </b-form>
+    </b-modal>
+  </ValidationObserver>
+</template>
+
+<script>
+import InputWidget from "../../../core/components/input-widget/InputWidget";
+import ArticleFormModel from "./ArticleFormModel";
+import {createNamespacedHelpers} from "vuex";
+
+const {mapState, mapActions} = createNamespacedHelpers('article');
+const {mapState: mapWorkspaceState} = createNamespacedHelpers('workspace');
+
+export default {
+  name: "ArticleForm",
+  components: {InputWidget},
+  data() {
+    return {
+      model: new ArticleFormModel(),
+      action: '',
+      resource: '',
+    }
+  },
+  computed: {
+    ...mapState(['showModal', 'modalArticle', 'currentArticle', 'isArticle']),
+    ...mapWorkspaceState(['currentWorkspace']),
+    modalTitle() {
+      if (this.isArticle) {
+        if (this.modalArticle) {
+          return this.$t(`Update article '{name}'`, {name: this.modalArticle.title});
+        } else {
+          return this.$t(`Create new article`);
+        }
+      } else {
+        if (this.modalArticle) {
+          return this.$t(`Update folder '{name}'`, {name: this.modalArticle.title});
+        } else {
+          return this.$t(`Create new folder`);
+        }
+      }
+    }
+  },
+  watch: {
+    modalArticle() {
+      if (this.modalArticle) {
+        this.model = new ArticleFormModel(this.modalArticle);
+      }
+    },
+  },
+  methods: {
+    ...mapActions(['hideArticleModal', 'createArticle', 'updateArticle', 'getArticlesByParent', 'getArticlesByWorkspace']),
+    async onSubmit() {
+      this.resource = 'folder';
+      if (this.currentWorkspace.id) {
+        this.model.workspace_id = this.currentWorkspace.id;
+      }
+      this.model.article_id = this.currentArticle.id;
+      this.model.isArticle = this.isArticle;
+
+      if (this.isArticle) {
+        this.resource = 'article';
+      } else {
+        this.resource = 'folder';
+      }
+      let res
+      if (this.model.id) {
+        this.action = 'updated';
```

Comment:
@skavt We do not need action variable in data, we can declare local variable here.